lectures/5-commitments.tex

```diff
@@ -0,0 +1,159 @@
+EC point $U$ is chosen randomly using "Nothing-up-my-sleeve" to assure no one knows the discrete logarithm of a selected point.
+
+\begin{remark}
+    \textbf{Transparent random points generation}
+
+    User can pick the publicly chosen random number (like a hash of project name, first numbers of $\pi$, etc), and hash that result to obtain another value. If that results in an $x$ value that lies on the elliptic curve, use that as the random point and hash the $(x, y)$ pair again (to obtain the next one, it needed). Otherwise, if the $x$-value does not land on the curve, increment $x$ until it does. Because the committer is not generating the points, they don`t know their discrete log. 
+\end{remark}
+
+Pedersen commitment scheme algorithm:
+\begin{enumerate}
+    \item Prover and Verifier agrees on $G$ and $U$ points
+    \item Prover selects a value $m$ to commit and a blinder factor $r$:
+            $m \leftarrow \mathbb{Z}$,
+            $r \xleftarrow{R} \mathbb{Z}$
+    \item Prover generates a commitment and sends it to the Verifier:
+            $c = mG + rU$
+\end{enumerate}
+
+During the verification stage, prover reveals $(m, r)$ to the verifier. 
+To check the commitment, verifier computes: $c_1 = mG + rU$.
+
+If $c_1 = c$, prover has revealed the correct pair $(m, r)$.
+
+\begin{remark}
+    In case the discrete logarithm of $U$ is leaked, the \textit{binding} property can be violated by the \textit{Prover}:
+
+    $c = mG + rU = m \cdot G + r*u \cdot G = (m + r*u) \cdot G$
+    
+    For example, $(m + u, r - 1)$ will have the same commitment value:
+
+    $(m+u + (r-1)*u) \cdot G = (m + u - u + r*u) \cdot G = (m + r*u) \cdot G$ 
+        
+\end{remark}
+
+\textbf{Commitment aggregation}
+
+Pedersen commitment have some advantages compared to hash-based commitments.
+Additively homomorphic property allows to accumulate multiple commitments into one.
+Consider two pairs: $(m_1, r_1), (m_2, r_2)$.
+
+\begin{center}
+    $C_1 = m_1G + r_1U$,
+
+    $C_2 = m_2G + r_2U$,
+
+    $C_a = C_1 + C_2 = (m_1 + m_2)G + (r_1 + r_2)U$
+\end{center}
+
+This works for any number of commitments, so we can encode as many points as we like in a single one.
+
+\vspace{0.5 cm}
+
+\subsubsection{Vector commitments}
+
+Vector commitment schemes allows to commit to a vector of values rather than a value and a blinding term.
+
+\vspace{0.5 cm}
+
+\textbf{Pedersen Vector Commitments}
+
+Suppose we have a set of random elliptic curve points $(G_1,...,G_n)$ of group $\mathbb{G}$ 
+(that we do not know the discrete logarithm of), a vector $[m_1, m_2 ... m_n]$ and a random value $r$. We can do the following:
```

Comment:
For triple dots, use `\dots` or `\ldots`

Reply:
To denote a vector, do not use square brackets. That is, it is generally better to write $(m_1,m_2,\dots,m_n)$